urllib3/connectionpool.py

```diff
@@ -568,25 +590,19 @@ def urlopen(self, method, url, body=None, headers=None, retries=None,
             # Close the connection. If a connection is reused on which there
             # was a Certificate error, the next request will certainly raise
             # another Certificate error.
-            if conn:
-                conn.close()
-                conn = None
+            conn = self._cleanup_conn(conn, replace_conn=not release_conn)
             raise SSLError(e)
 
         except SSLError:
             # Treat SSLError separately from BaseSSLError to preserve
             # traceback.
-            if conn:
-                conn.close()
-                conn = None
+            conn = self._cleanup_conn(conn, replace_conn=not release_conn)
             raise
 
         except (TimeoutError, HTTPException, SocketError, ConnectionError) as e:
-            if conn:
-                # Discard the connection for these exceptions. It will be
-                # be replaced during the next _get_conn() call.
-                conn.close()
-                conn = None
+            # Discard the connection for these exceptions. It will be
+            # be replaced during the next _get_conn() call.            
+            conn = self._cleanup_conn(conn, replace_conn=not release_conn)
```

Comment:
Yes, but only if release_conn is False. The exception is raised further down the line when retries are all used up. If a connection is closed and no response will be consumed, and if release_conn is False, when else would it be release?